src/quote/getQuoteAndBuyCoverInputs.ts

```diff
@@ -0,0 +1,161 @@
+import axios, { AxiosError } from 'axios';
+import dotenv from 'dotenv';
+
+import { calculatePremiumWithCommissionAndSlippage } from '../buyCover/calculatePremiumWithCommissionAndSlippage';
+import {
+  CoverAsset,
+  CoverId,
+  DEFAULT_COMMISSION_RATIO,
+  DEFAULT_SLIPPAGE,
+  MINIMUM_COVER_PERIOD,
+  NEXUS_MUTUAL_DAO_TREASURY_ADDRESS,
+  TARGET_PRICE_DENOMINATOR,
+} from '../constants/buyCover';
+import {
+  Address,
+  CoverRouterQuoteResponse,
+  ErrorApiResponse,
+  GetQuoteApiResponse,
+  GetQuoteResponse,
+  IntString,
+  Integer,
+  PoolCapacity,
+} from '../types';
+
+dotenv.config();
+
+type CoverRouterQuoteParams = {
+  productId: Integer;
+  amount: IntString;
+  period: Integer;
+  coverAsset: CoverAsset;
+};
+
+async function getQuoteAndBuyCoverInputs(
+  productId: Integer,
+  coverAmount: IntString,
+  coverPeriod: Integer, // days
+  coverAsset: CoverAsset,
+  coverBuyerAddress: Address,
+  paymentAsset: CoverAsset = coverAsset,
+  slippage: number = DEFAULT_SLIPPAGE, // 0.1%
+  ipfsData: string = '', // IPFS base32 hash value
+): Promise<GetQuoteApiResponse | ErrorApiResponse> {
+  let result: GetQuoteResponse | undefined = undefined;
+
+  if (!process.env.COVER_ROUTER_URL) {
+    return { result, error: { message: 'Missing COVER_ROUTER_URL env var', data: {} } };
+  }
+
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return { result, error: { message: 'Invalid productId: must be a positive integer', data: {} } };
+  }
+
+  if (!/^\d+$/.test(coverAmount) || parseInt(coverAmount, 10) <= 0) {
+    return { result, error: { message: 'Invalid coverAmount: must be a positive integer string', data: {} } };
+  }
+
+  if (!Number.isInteger(coverPeriod) || coverPeriod < MINIMUM_COVER_PERIOD) {
+    return {
+      result,
+      error: { message: `Invalid coverPeriod: must be at least ${MINIMUM_COVER_PERIOD} days`, data: {} },
+    };
+  }
+
+  if (!Object.values(CoverAsset).includes(coverAsset)) {
+    return { result, error: { message: 'Invalid coverAsset', data: {} } };
+  }
+
+  if (!Object.values(CoverAsset).includes(paymentAsset)) {
```

Comment:
For payment asset, we also have `NXM = 255`  as value in the enum - not for cover asset, that one is ok.

Reply:
Also worth noting that we allow as payment asset only the selected cover asset and `NXM`.

Reply:
> Also worth noting that we allow as payment asset only the selected cover asset and NXM

since CoverBroker does not support NXM payments, we should probably not include NXM as a payment option in the SDK too.

with that in mind, it might be best to remove `paymentAsset` in the params as the only valid value for is `coverAsset`.